lib/charms/layer/apache_kafka.py

```diff
@@ -14,14 +16,29 @@ def __init__(self, dist_config=None):
         self.resources = {
             'kafka': 'kafka-%s' % utils.cpu_arch(),
         }
-        self.verify_resources = utils.verify_resources(*self.resources.values())
+
+    def verify_resources(self):
+        if hookenv.resource_get(self.resources['kafka']):
+            return True
+        else:
+            return utils.verify_resources(*self.resources.values())()
 
     def install(self):
         self.dist_config.add_users()
         self.dist_config.add_dirs()
-        jujuresources.install(self.resources['kafka'],
-                              destination=self.dist_config.path('kafka'),
-                              skip_top_level=True)
+        filename = hookenv.resource_get(self.resources['kafka'])
```

Comment:
I liked this fallback mechanism. I had similar in the etcd charm.  The one thing I did notice is that it made it less clear where the package was coming from. I feel like we need a way to report this back to us (the charm author) when running support.

Perhaps a state or an action to generate this information would be helpful